Bitbucket.Authentication/Authority.cs

```diff
@@ -77,7 +74,35 @@ public string UserUrl
                 OAuth.OAuthAuthenticator oauth = new OAuth.OAuthAuthenticator();
                 try
                 {
-                    return await oauth.GetAuthAsync(targetUri, scope, CancellationToken.None);
+                    var result = await oauth.GetAuthAsync(targetUri, scope, CancellationToken.None);
+
```

Comment:
If the OAuth dance is successful then call the /user REST api so we can cross check the supplied username and the one stored by Bitbucket